src/veritysetup/veritysetup.c

```diff
@@ -109,7 +140,49 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "hash-offset="))) {
+                else if ((val = startswith(word, "superblock="))) {
+
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+
+                        arg_superblock = r;
+                } else if ((val = startswith(word, "format="))) {
+                        int i;
+
+                        r = safe_atoi(val, &i);
```

Comment:
The `cryptsetup` handles the format/hash argument as an [integer](https://gitlab.com/cryptsetup/cryptsetup/-/blob/main/lib/libcryptsetup.h#L505).

So I guess it worth sticking to parse-and-store it as a integer instead of parsing-and-allocating it as string to parse it later as an integer for `cryptsetup`. Furthermore, if allocation is needed for string, it would complicate the freeing of mermore if mulitiple `format=` is given as options (see `salt=` parsing).

Following the [comment](https://github.com/systemd/systemd/pull/18789#discussion_r938611545), I may remove the checking for the supported current values (`0` and `1`).

Reply:
I mean, you filter values > 1 anyway here. And accidentally allow values < 0 i guess...

So why not:

```c
if (!STR_IN_SET(val, "0", "1")) 
        return log_error_errno(SYNTHETIC_ERRNO(val), "Unknown format: %s", val);
arg_format = val[0] - '0';
```

shorter, safer, and easier on the eye I think

Reply:
Ok, I will do.